### PR TITLE
feat(StatCard): add StatCard component with delta auto-inference and CSS-var theming

### PR DESCRIPTION
**Pull request by murdore**
## What

New `StatCard` primitive component under `src/lib/StatCard/`.

**Props** (all optional — fully backward-compatible new component):
- `title?`, `value?`, `delta?`, `deltaPositive?`, `subtitle?` — display fields
- `footer?: Snippet`, `valueSnippet?: Snippet` — slot-like composition
- `testId?`, `classes?` — standard testability / theming hooks
- `onclick?: (event: MouseEvent) => void` — interactive-card mode (adds `role="button"`, `tabindex=0`, Enter/Space keyboard support)

**Delta auto-inference**: when `deltaPositive` is not provided, the sign of the `delta` string (`+` / `-`) drives the positive/negative colour class automatically.

**CSS theming**: `--statcard-*` CSS variables with light defaults cover layout (gap, padding, border, border-radius, box-shadow, width/height), per-sub-element colours/sizes, and interactive focus-ring.

## Why / conventions aligned

- `MandatoryStatCardProperties = Record<string, never>` sentinel matches the Card pattern
- `StatCardEventProperties` separate type matches the Banner/Pill convention
- `handleKeydown` uses `function` keyword matching Card.svelte's local handler style
- Sub-element `data-pw` removed — only root element carries `data-pw`, consistent with every other library component (Card, Banner, Badge, etc.)
- `null` used instead of `undefined` in ternary expressions (project ESLint `no-restricted-syntax` rule)

## Test plan

- [ ] Import `StatCard` from `@juspay/svelte-ui-components` in a consumer
- [ ] Render with `title`, `value`, `delta="+12.5%"` — verify green delta colour auto-inferred
- [ ] Render with `delta="-3.2%"` — verify red delta colour
- [ ] Render with `delta="0%"` — verify neutral (no colour class)
- [ ] Render with `deltaPositive={false}` and `delta="+5%"` — override wins (red)
- [ ] Render with `onclick` — verify `role="button"`, `tabindex=0`, Enter/Space fires click
- [ ] Render without `onclick` — verify no interactive attributes
- [ ] Render with `footer` and `valueSnippet` snippets — verify custom rendering
- [ ] Verify CSS vars (`--statcard-background`, `--statcard-value-color`, etc.) are overridable

## Summary by CodeRabbit

* **New Features**
  * Added StatCard component displaying title, value, optional delta indicator, subtitle, and footer elements.
  * Supports keyboard interaction and custom value rendering.
  * Themeable via CSS variables.
  * Available as both Svelte component and Web Component (`<sui-stat-card>`).

* **Documentation**
  * Added comprehensive StatCard documentation with API reference, usage examples, and Web Component integration guide.

### CODE REVIEW

File: src/lib/StatCard/StatCard.svelte
Comment (Tara-ag) [inline src/lib/StatCard/StatCard.svelte:1]: 💡 MINOR: Missing documentation file (`docs/StatCard.md`)

All components in this library have corresponding documentation in the `docs/` folder (e.g., `docs/Card.md`, `docs/Banner.md`). The PR should include a `docs/StatCard.md` file documenting:
- Usage examples
- Props table
- Events table  
- CSS variables reference
- Web component tag information

This ensures consumers can discover and properly use the component.

Comment (Tara-ag) [inline src/lib/StatCard/StatCard.svelte:1]: 💡 MINOR: Missing web component wrapper for WC build consistency

The library maintains web component wrappers in `src/wc/components/` for all exported components (see `Card.wc.svelte`, `Banner.wc.svelte`, etc.). StatCard needs a corresponding `StatCard.wc.svelte` wrapper and an import in `src/wc/index.ts` to maintain parity with the rest of the library's web component build.

Suggested addition to `src/wc/components/StatCard.wc.svelte`:
```svelte
<svelte:options
  customElement={{
    tag: 'sui-statcard',
    shadow: 'open',
    props: {
      title: { type: 'String', reflect: true },
      value: { type: 'String', reflect: true },
      delta: { type: 'String', reflect: true },
      deltaPositive: { type: 'Boolean', reflect: true },
      subtitle: { type: 'String', reflect: true },
      classes: { type: 'String' },
      testId: { type: 'String' }
    }
  }}
/>

<script lang="ts">
  import StatCard from '$lib/StatCard/StatCard.svelte';
  let props = $props();
</script>

<StatCard {...props} />
```

And add to `src/wc/index.ts`:
```typescript
import './components/StatCard.wc.svelte';
```

**Review by Tara-ag — APPROVED** [review]
## Review Summary

**Files reviewed:** 3 (1 new component + types + export)
**New issues raised:** 2 MINOR

### Assessment

The StatCard component is well-implemented and follows established library patterns:

✅ **Component structure** mirrors `Card.svelte` — consistent interactive element handling with `role="button"`, `tabindex`, and Enter/Space keyboard support  
✅ **Accessibility** follows the same `svelte-ignore a11y_no_noninteractive_tabindex` pattern used across the library (Card, Pill, Book, KeyboardInput, etc.)  
✅ **Svelte 5 reactivity** correctly uses `$props()`, `$derived()` for computed states  
✅ **CSS custom properties** follow the `--statcard-*` naming convention with sensible defaults  
✅ **Type definitions** follow the `Mandatory/Optional/Event` pattern used by Banner/Pill/Card  
✅ **Public API** properly exported in `src/lib/index.ts` — no breaking changes, purely additive  
✅ **Delta auto-inference** logic is sound (handles `+`/`-` prefix detection with explicit override support)

### Minor Consistency Gaps (non-blocking)

1. **Missing web component wrapper** — `src/wc/components/StatCard.wc.svelte` and entry in `src/wc/index.ts` needed for WC build parity
2. **Missing documentation** — `docs/StatCard.md` should be added following the Card.md template structure

These are documentation/build-consistency items that don't affect runtime behavior or public API correctness. Recommend addressing in a follow-up or before merge if the maintainer prefers complete parity.

**Verdict:** Approve — the component is production-ready and follows all critical library conventions.

File: src/wc/components/StatCard.wc.svelte [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  customElement={{
+    tag: 'sui-stat-card',
+    shadow: 'open',
+    props: {
+      title: { type: 'String', reflect: true },
+      value: { type: 'String', reflect: true },
+      delta: { type: 'String', reflect: true },
+      deltaPositive: { type: 'Boolean', attribute: 'delta-positive' },
+      subtitle: { type: 'String', reflect: true },
+      classes: { type: 'String' },
```
Comment (Tara-ag) [inline src/wc/components/StatCard.wc.svelte:11]: 💡 MINOR: Missing `onclick` prop in web component definition

The `StatCard` component supports an `onclick` handler that makes the card interactive (adding `role="button"`, `tabindex`, and keyboard support). However, this prop is not declared in the web component's `props` configuration, which means consumers using the web component (`<sui-stat-card>`) cannot utilize the interactive mode.

For consistency with other interactive components like `Button.wc.svelte` and `Pill.wc.svelte`, add:
```typescript
onclick: { type: 'Object' }
```

This allows the web component to pass through the click handler to the underlying Svelte component.

**Review by Tara-ag — COMMENTED** [review]
## Review Summary

**Files reviewed:** 7 (1 new component + types + docs + WC wrapper + exports)
**New issues raised:** 1 MINOR

### Assessment

The StatCard component is well-implemented and follows established library conventions:

✅ **Component structure** mirrors `Card.svelte` — consistent interactive element handling with `role="button"`, `tabindex`, and Enter/Space keyboard support  
✅ **Accessibility** follows the same `svelte-ignore a11y_no_noninteractive_tabindex` pattern used across the library  
✅ **Svelte 5 reactivity** correctly uses `$props()`, `$derived()` for computed states  
✅ **CSS custom properties** follow the `--statcard-*` naming convention with sensible defaults  
✅ **Type definitions** follow the `Mandatory/Optional/Event` pattern used by Banner/Pill/Card  
✅ **Public API** properly exported in `src/lib/index.ts` — no breaking changes, purely additive  
✅ **Delta auto-inference** logic is sound (handles `+`/`-` prefix detection with explicit override support)  
✅ **Documentation** comprehensive `docs/StatCard.md` with usage examples, props table, events, and CSS variables  
✅ **Web component wrapper** present and registered in `src/wc/index.ts`

### Minor Issue

1. **Missing `onclick` prop in WC definition** — The web component wrapper doesn't declare the `onclick` prop, preventing consumers from using the interactive mode via the web component tag. See inline comment for suggested fix.

**Verdict:** The component is production-ready. The missing `onclick` in the WC wrapper is a minor gap that should be addressed for full feature parity between Svelte and web component consumers.

File: src/wc/components/StatCard.wc.svelte [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      onclick: { type: 'Object' }
+    }
+  }}
+/>
+
+<script lang="ts">
+  import StatCard from '$lib/StatCard/StatCard.svelte';
+  let props = $props();
+</script>
+
```
Comment (Tara-ag) [inline src/wc/components/StatCard.wc.svelte:22]: 💡 MINOR: Missing snippet slot forwarding for `footer` and `valueSnippet`

The web component wrapper doesn't forward the `footer` and `valueSnippet` snippets as slots, which means web component consumers cannot use these composition features. Compare with `Banner.wc.svelte` which properly forwards snippets like `{#snippet icon()}<slot name="icon"></slot>{/snippet}`.

Without this, the following web component usage won't work:
```html
<sui-stat-card title="Orders" value="1,482">
  <!-- footer snippet has no slot to render into -->
</sui-stat-card>
```

**Suggested fix:**
```svelte
<StatCard {...props}>
  {#snippet footer()}
    <slot name="footer"></slot>
  {/snippet}
  {#snippet valueSnippet()}
    <slot name="value-snippet"></slot>
  {/snippet}
</StatCard>
```

This ensures parity between Svelte and web component consumers for the `footer` and `valueSnippet` composition features.

File: src/lib/StatCard/StatCard.svelte [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+      if (event.currentTarget instanceof HTMLElement) {
+        event.currentTarget.click();
+      }
+    }
+  }
+</script>
+
+<!-- svelte-ignore a11y_no_noninteractive_tabindex -->
+<div
+  class="statcard {classes ?? ''}"
```
Comment (Tara-ag) [inline src/lib/StatCard/StatCard.svelte:52]: 💬 SUGGESTION: Consider adding `aria-pressed` for toggle-like interactive cards

When `onclick` is provided, the card becomes a button. If the card represents a toggleable state (e.g., selected/unselected metric view), consider adding `aria-pressed` support. This is optional since the component doesn't inherently track pressed state, but worth documenting if consumers implement such patterns.

Not blocking - the current implementation correctly follows the `Card.svelte` pattern for interactive elements.

File: src/lib/StatCard/StatCard.svelte [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+  const isInteractive = $derived(typeof onclick === 'function');
+
+  /**
+   * When `deltaPositive` is explicitly provided use it; otherwise infer from
+   * the leading character of `delta` — '+' → positive, '-' → negative,
+   * anything else (e.g. '0%') → undefined (neutral, no colour applied).
+   */
+  const resolvedDeltaPositive = $derived(
+    typeof deltaPositive === 'boolean'
+      ? deltaPositive
```
Comment (Tara-ag) [inline src/lib/StatCard/StatCard.svelte:26]: 💡 MINOR: Delta inference logic edge case with whitespace-only strings

The current logic treats a whitespace-only string like `"   "` as having a delta (since `trimStart().startsWith()` would fail but `length > 0` passes), but it wouldn't get any color class. Consider trimming before the length check:

```typescript
const resolvedDeltaPositive = $derived(
  typeof deltaPositive === 'boolean'
    ? deltaPositive
    : typeof delta === 'string' && delta.trim().length > 0
      ? delta.trimStart().startsWith('+')
        ? true
        : delta.trimStart().startsWith('-')
          ? false
          : null
      : null
);
```

Also update `hasDelta` similarly:
```typescript
const hasDelta = $derived(typeof delta === 'string' && delta.trim().length > 0);
```

This ensures whitespace-only deltas are treated consistently as "no delta" rather than rendering an empty/neutral badge.

**Review by Tara-ag — APPROVED** [review]
## Review Summary

**Files reviewed:** 7 (1 new component + types + docs + WC wrapper + exports)
**New issues raised:** 3 (2 MINOR, 1 SUGGESTION)

### Assessment

The StatCard component is well-implemented and follows established library conventions:

✅ **Component structure** mirrors `Card.svelte` — consistent interactive element handling with `role="button"`, `tabindex`, and Enter/Space keyboard support  
✅ **Accessibility** follows the same `svelte-ignore a11y_no_noninteractive_tabindex` pattern used across the library  
✅ **Svelte 5 reactivity** correctly uses `$props()`, `$derived()` for computed states  
✅ **CSS custom properties** follow the `--statcard-*` naming convention with sensible defaults  
✅ **Type definitions** follow the `Mandatory/Optional/Event` pattern used by Banner/Pill/Card  
✅ **Public API** properly exported in `src/lib/index.ts` — no breaking changes, purely additive  
✅ **Delta auto-inference** logic is sound (handles `+`/`-` prefix detection with explicit override support)  
✅ **Documentation** comprehensive `docs/StatCard.md` with usage examples, props table, events, and CSS variables  
✅ **Web component wrapper** present and registered in `src/wc/index.ts` with `onclick` prop included

### Issues Raised

1. **MINOR:** Web component wrapper missing snippet slot forwarding for `footer` and `valueSnippet` — web component consumers cannot use these composition features without slot forwarding (see `Banner.wc.svelte` for the pattern)

2. **MINOR:** Delta inference edge case with whitespace-only strings — consider using `delta.trim().length > 0` instead of `delta.length > 0` to treat whitespace-only strings as "no delta"

3. **SUGGESTION:** Consider documenting `aria-pressed` pattern if consumers implement toggle-like behavior with the interactive card

### Verdict

**Approve** — The component is production-ready. The issues raised are minor polish items that don't affect runtime behavior or public API correctness. The PR demonstrates excellent adherence to library patterns and conventions.

**Comment by murdore** [comment]
Rebased onto current \`release\` (was 38 commits behind — this resolves the stale version/CSS-var diff comments, which were merge-base artifacts). Addressed:

- **WC snippet forwarding**: Added `{#snippet footer()}<slot name="footer"></slot>{/snippet}` and `{#snippet valueSnippet()}<slot name="value-snippet"></slot>{/snippet}` to `StatCard.wc.svelte` so web component consumers can use the footer and valueSnippet composition features (matching the `Banner.wc.svelte` pattern).
- **Delta whitespace edge case**: Changed `delta.length > 0` → `delta.trim().length > 0` in both `resolvedDeltaPositive` and `hasDelta` so whitespace-only strings are treated as "no delta" rather than rendering an empty neutral badge.

Intentionally not changed:
- **`aria-pressed` support** (reviewer suggestion): The component does not track toggled state — adding `aria-pressed` without state management would produce incorrect semantics. Consumers needing toggle behaviour can wire this via the `classes` prop and their own state.
- **Missing docs/wc wrapper comments**: These were stale — `docs/StatCard.md`, `StatCard.wc.svelte`, `onclick` in wc props, and all registry entries were already present in the PR.

File: src/wc/components/StatCard.wc.svelte
Comment (Tara-ag) [inline src/wc/components/StatCard.wc.svelte:1]: 💬 SUGGESTION: Snippet slot forwarding is now correctly implemented

The `footer` and `valueSnippet` snippets are properly forwarded as slots, matching the `Banner.wc.svelte` pattern. This resolves the previously raised concern about web component consumers being unable to use these composition features.

The slot names follow kebab-case convention (`value-snippet` vs `valueSnippet`) which is consistent with other web component wrappers in the library.

File: src/lib/StatCard/StatCard.svelte (hunk 4 above)
Comment (Tara-ag) [inline src/lib/StatCard/StatCard.svelte:26]: 💬 SUGGESTION: Delta whitespace handling is correctly implemented

The current implementation properly handles whitespace-only strings by using `delta.trim().length > 0` in both `resolvedDeltaPositive` and `hasDelta` derivations. This ensures whitespace-only strings like `"   "` are treated as "no delta" rather than rendering an empty/neutral badge.

The logic correctly:
1. Trims before checking length
2. Uses `trimStart()` for sign detection (handles cases like `"  +12%"`)
3. Returns `null` for neutral/undetermined states

File: src/lib/StatCard/StatCard.svelte [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+      if (event.currentTarget instanceof HTMLElement) {
+        event.currentTarget.click();
+      }
+    }
+  }
+</script>
+
+<!-- svelte-ignore a11y_no_noninteractive_tabindex -->
+<div
+  class="statcard {classes ?? ''}"
```
Comment (Tara-ag) [inline src/lib/StatCard/StatCard.svelte:52]: 💬 SUGGESTION: Interactive card accessibility pattern

The `role="button"`, `tabindex="0"`, and `handleKeydown` implementation correctly follows the `Card.svelte` pattern for interactive elements. The Enter/Space keyboard support is properly implemented with `event.preventDefault()` for Space to avoid page scrolling.

Consider documenting that consumers implementing toggle-like behavior should manage `aria-pressed` state externally, as this component intentionally doesn't track pressed state internally (consistent with the `Card` component design).

File: src/lib/index.ts
Comment (Tara-ag) [inline src/lib/index.ts:1]: 💬 SUGGESTION: Export placement and ordering

The `StatCard` export is correctly placed alphabetically among the chart components (after `SankeyChart`). The type export is also correctly added at the end of the type exports section. This maintains consistency with the existing organization pattern in the barrel file.

Both the component and its types are properly exported for library consumers.

**Review by Tara-ag — APPROVED** [review]
## Review Summary

**Files reviewed:** 7 (1 new component + types + docs + WC wrapper + exports + index updates)
**New issues raised:** 0 (all previously identified issues have been resolved)

### Assessment

The StatCard component is well-implemented and follows all established library conventions:

✅ **Component structure** mirrors `Card.svelte` — consistent interactive element handling with `role="button"`, `tabindex`, and Enter/Space keyboard support  
✅ **Accessibility** follows the same `svelte-ignore a11y_no_noninteractive_tabindex` pattern used across the library  
✅ **Svelte 5 reactivity** correctly uses `$props()`, `$derived()` for computed states with proper cleanup  
✅ **CSS custom properties** follow the `--statcard-*` naming convention with sensible defaults  
✅ **Type definitions** follow the `Mandatory/Optional/Event` pattern used by Banner/Pill/Card  
✅ **Public API** properly exported in `src/lib/index.ts` — no breaking changes, purely additive  
✅ **Delta auto-inference** logic correctly handles whitespace with `trim().length > 0`  
✅ **Documentation** comprehensive `docs/StatCard.md` with usage examples, props table, events, and CSS variables  
✅ **Web component wrapper** present in `src/wc/components/StatCard.wc.svelte` with:
   - Proper snippet slot forwarding for `footer` and `valueSnippet`
   - `onclick` prop declaration for interactive mode
   - Registration in `src/wc/index.ts`  
✅ **SSR safety** — no unguarded `window`/`document`/`localStorage` access at module or component init  

### Previously Raised Issues — All Resolved

1. ✅ **Documentation** — `docs/StatCard.md` added with full API reference
2. ✅ **Web component wrapper** — Created with proper slot forwarding pattern matching `Banner.wc.svelte`
3. ✅ **WC `onclick` prop** — Added to custom element props definition
4. ✅ **Snippet slot forwarding** — `footer` and `valueSnippet` properly forwarded as slots
5. ✅ **Delta whitespace handling** — Uses `delta.trim().length > 0` to treat whitespace-only strings as "no delta"

### Verdict

**Approve** — The component is production-ready. All previously identified issues have been addressed. The PR demonstrates excellent adherence to library patterns and conventions. The new component is fully backward compatible (purely additive) with no breaking changes to existing consumers.

**Comment by murdore** [comment]
Rebased onto release and addressed all actionable review comments.

**Addressed:**
- **WC snippet slot forwarding** (`StatCard.wc.svelte`): Added `{#snippet footer()}<slot name="footer"></slot>{/snippet}` and `{#snippet valueSnippet()}<slot name="value-snippet"></slot>{/snippet}` matching the `Banner.wc.svelte` pattern — web component consumers can now use these composition features.
- **Delta whitespace handling** (`StatCard.svelte`): Changed `delta.length > 0` → `delta.trim().length > 0` in both `resolvedDeltaPositive` and `hasDelta` derivations, so whitespace-only strings (e.g. `"   "`) are treated as "no delta" rather than rendering an empty neutral badge.
- **Arrow function style** (`handleKeydown`): Converted from `function` declaration to arrow function `const handleKeydown = (...) =>` to match the `Card.svelte` pattern that StatCard mirrors.

**Skipped (genuinely already done / informational):**
- **docs/StatCard.md** — already present in the branch from the previous pass.
- **WC wrapper `onclick` prop** — already present (`onclick: { type: 'Object' }`).
- **aria-pressed SUGGESTION** — intentional design decision; documented in the JSDoc/docs that consumers manage this externally.
- **Snippet forwarding SUGGESTION follow-up** and **delta whitespace SUGGESTION follow-up** — these were Tara-ag's positive follow-up confirmations on work already done, not new action items.

Gate: `check` (0 errors) ✅ · `lint` (prettier + eslint) ✅ · `build` (svelte-package) ✅

File: src/wc/components/StatCard.wc.svelte
Comment (Tara-ag) [inline src/wc/components/StatCard.wc.svelte:1]: 💡 MINOR: Boolean prop `deltaPositive` missing `reflect: true`

The `deltaPositive` boolean prop in the web component definition doesn't have `reflect: true`, unlike other boolean props in the library (e.g., `dismissible` in `Pill.wc.svelte`). This means the attribute won't be reflected to the DOM when the property changes programmatically.

**Suggested fix:**
```typescript
deltaPositive: { type: 'Boolean', attribute: 'delta-positive', reflect: true }
```

This ensures consistency with other boolean props in the library's web component wrappers.

**Review by Tara-ag — APPROVED** [review]
## Review Summary

**Files reviewed:** 7 (all new component files)

**Issues found:**
| Severity | Count | Description |
|----------|-------|-------------|
| 🔒 CRITICAL | 0 | None |
| ⚠️ MAJOR | 0 | None |
| 💡 MINOR | 1 | `deltaPositive` missing `reflect: true` in WC definition |
| 💬 SUGGESTION | 0 | None |

### Assessment

This PR introduces a well-designed new `StatCard` component that follows the library's established patterns:

✅ **Public API**: Clean props interface with proper TypeScript types, follows `Optional/Mandatory/Event` pattern
✅ **Accessibility**: Proper `role="button"`, `tabindex`, and Enter/Space keyboard support for interactive mode
✅ **SSR Safety**: No unguarded `window`/`document` access
✅ **CSS Theming**: Comprehensive CSS custom property coverage
✅ **Web Component**: Proper wrapper with snippet slot forwarding matching `Banner.wc.svelte` pattern
✅ **Documentation**: Complete `docs/StatCard.md` with props, events, and CSS variables tables
✅ **Exports**: Correctly added to `src/lib/index.ts` and `src/wc/index.ts`

### Minor Issue Raised
- `deltaPositive` boolean prop in WC definition should have `reflect: true` for consistency with other boolean props in the library

### Previous Review Items
All items from the previous review cycle have been addressed:
- ✅ Documentation file added
- ✅ Web component wrapper added
- ✅ `onclick` prop included in WC definition
- ✅ Snippet slot forwarding implemented
- ✅ Delta whitespace handling corrected

**Recommendation:** Approve with minor suggestion to add `reflect: true` to `deltaPositive`.

**Comment by murdore** [comment]
Latest review round addressed:

- **`deltaPositive: reflect: true` in WC definition** (comment #3423495633) — added `reflect: true` to the `deltaPositive` Boolean prop in `src/wc/components/StatCard.wc.svelte`, matching the pattern used by every other Boolean prop in the library (`dismissible`, `visible`, `disabled`, `checked`, etc.). This ensures the attribute is reflected to the DOM when the property changes programmatically.

Also rebased onto `release` (now includes 2.60.0 + Select itemTestId docs) — no conflicts, clean forward rebase.

Intentionally not changed (no action needed):

- **`aria-pressed` documentation suggestion** (comment #3422510355 / #3423255025) — this is a SUGGESTION, not a MINOR issue. The current implementation intentionally does not track pressed state (consistent with `Card.svelte`). The reviewer's own follow-up comment at 18:54:32 confirms the current pattern is correct and suggests documenting consumer-side usage externally, which is out of scope for this PR.
- All other items from prior rounds (docs, WC wrapper, onclick prop, snippet forwarding, delta whitespace) — already resolved in previous pushes, confirmed by the reviewer's 18:55:02 summary marking all issues as ✅.

File: src/wc/components/StatCard.wc.svelte [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  customElement={{
+    tag: 'sui-stat-card',
+    shadow: 'open',
+    props: {
+      title: { type: 'String', reflect: true },
+      value: { type: 'String', reflect: true },
+      delta: { type: 'String', reflect: true },
+      deltaPositive: { type: 'Boolean', attribute: 'delta-positive' },
+      subtitle: { type: 'String', reflect: true },
+      classes: { type: 'String' },
```
Comment (Tara-ag) [inline src/wc/components/StatCard.wc.svelte:11]: 💡 MINOR: Boolean prop `deltaPositive` missing `reflect: true`

The `deltaPositive` boolean prop in the web component definition doesn't have `reflect: true`, unlike other boolean props in the library (e.g., `dismissible` in `Pill.wc.svelte`, `checked` in `Checkbox.wc.svelte`, `disabled` in `Button.wc.svelte`). This means the attribute won't be reflected to the DOM when the property changes programmatically.

**Suggested fix:**
```typescript
deltaPositive: { type: 'Boolean', attribute: 'delta-positive', reflect: true }
```

This ensures consistency with other boolean props in the library's web component wrappers.

**Review by Tara-ag — COMMENTED** [review]
## Review Summary

**Files reviewed:** 7 (1 new component + types + docs + WC wrapper + exports)
**New issues raised:** 1 MINOR

### Assessment

The StatCard component is well-implemented and follows established library conventions:

✅ **Component structure** mirrors `Card.svelte` — consistent interactive element handling with `role="button"`, `tabindex`, and Enter/Space keyboard support  
✅ **Accessibility** follows the same `svelte-ignore a11y_no_noninteractive_tabindex` pattern used across the library  
✅ **Svelte 5 reactivity** correctly uses `$props()`, `$derived()` for computed states  
✅ **CSS custom properties** follow the `--statcard-*` naming convention with sensible defaults  
✅ **Type definitions** follow the `Mandatory/Optional/Event` pattern used by Banner/Pill/Card  
✅ **Public API** properly exported in `src/lib/index.ts` — no breaking changes, purely additive  
✅ **Delta auto-inference** logic correctly handles whitespace with `trim().length > 0`  
✅ **Documentation** comprehensive `docs/StatCard.md` with usage examples, props table, events, and CSS variables  
✅ **Web component wrapper** present in `src/wc/components/StatCard.wc.svelte` with:
   - Proper snippet slot forwarding for `footer` and `valueSnippet`
   - `onclick` prop declaration for interactive mode
   - Registration in `src/wc/index.ts`  
✅ **SSR safety** — no unguarded `window`/`document`/`localStorage` access at module or component init  

### Issue Raised

- **MINOR:** `deltaPositive` boolean prop in WC definition should have `reflect: true` for consistency with other boolean props in the library (e.g., `dismissible` in `Banner.wc.svelte`, `checked` in `Checkbox.wc.svelte`)

### Previous Review Items
All items from previous review cycles have been addressed:
- ✅ Documentation file added
- ✅ Web component wrapper added
- ✅ `onclick` prop included in WC definition
- ✅ Snippet slot forwarding implemented
- ✅ Delta whitespace handling corrected

**Recommendation:** Address the `reflect: true` consistency issue, then this PR is ready to merge. The component is production-ready and demonstrates excellent adherence to library patterns.

**Comment by murdore** [comment]
Latest review round addressed:

- **`deltaPositive` missing `reflect: true` in WC definition** — Added `reflect: true` to `deltaPositive: { type: 'Boolean', attribute: 'delta-positive', reflect: true }` in `src/wc/components/StatCard.wc.svelte`, consistent with all other boolean props across the library (`dismissible` in Banner/Pill, `checked`/`disabled`/`indeterminate` in Checkbox, etc.).

All previous review items were already resolved on the prior push (docs, WC wrapper, `onclick` prop, snippet slot forwarding, delta whitespace trimming). Gate: `pnpm run check` → 0 errors, `pnpm run lint` → clean, `pnpm run build` → clean.

File: src/lib/StatCard/StatCard.svelte
Comment (Tara-ag) [inline src/lib/StatCard/StatCard.svelte:1]: 💡 MINOR: Missing test coverage for StatCard component

The PR introduces a new component with complex logic (delta auto-inference, interactive keyboard handling) but does not include any tests. Per the project standards, new behavior should be covered by Vitest unit tests or Playwright integration tests.

Consider adding:
1. Unit tests for `resolvedDeltaPositive` derivation logic (positive/negative/neutral inference)
2. Unit tests for `hasDelta` with various edge cases (empty strings, whitespace-only)
3. Integration tests for keyboard interaction (Enter/Space triggering click)
4. Visual regression tests for delta color classes

This ensures the component's behavior remains stable across future changes.

**Review by Tara-ag — APPROVED** [review]
## Review Summary

**Files reviewed:** 7 changed files
- `docs/StatCard.md` (new)
- `docs/_index.json`
- `src/lib/StatCard/StatCard.svelte` (new)
- `src/lib/StatCard/properties.ts` (new)
- `src/lib/index.ts`
- `src/wc/components/StatCard.wc.svelte` (new)
- `src/wc/index.ts`

**Issues raised:** 1 MINOR

### Assessment

This PR introduces a well-designed `StatCard` component that follows the library's established patterns:

✅ **Public API & Backward Compatibility:** New component with all-optional props — fully backward-compatible addition
✅ **Accessibility:** Proper `role="button"`, `tabindex="0"`, and Enter/Space keyboard support for interactive mode
✅ **Svelte 5 Reactivity:** Correct use of `$derived` for `resolvedDeltaPositive`, `hasDelta`, and `isInteractive`
✅ **SSR Safety:** No unguarded `window`/`document` access at module/component init
✅ **Web Component Parity:** Properly exported in `src/wc/index.ts` with snippet slot forwarding
✅ **Documentation:** Comprehensive `docs/StatCard.md` with props table, CSS variables, and usage examples
✅ **Export Placement:** Correctly placed alphabetically in `src/lib/index.ts`

### Previously Raised Issues (All Resolved)
- ✅ `deltaPositive` now has `reflect: true` in WC definition
- ✅ Snippet slot forwarding for `footer` and `valueSnippet` implemented
- ✅ Delta whitespace handling correctly uses `.trim()`

### Minor Concern
- 💡 Missing test coverage for the new component (delta inference logic, keyboard interaction)

**Recommendation:** Approve — the component is production-ready. Consider adding tests in a follow-up PR to ensure long-term stability.

File: src/wc/components/StatCard.wc.svelte (hunk 2 above)
Comment (murdore) [reply src/wc/components/StatCard.wc.svelte:22]: Fixed in the current HEAD: `StatCard.wc.svelte` now forwards both `footer` and `valueSnippet` as named slots, matching the `Banner.wc.svelte` pattern. Resolving.

File: src/lib/StatCard/StatCard.svelte (hunk 4 above)
Comment (murdore) [reply src/lib/StatCard/StatCard.svelte:26]: Fixed in the current HEAD: both `resolvedDeltaPositive` and `hasDelta` use `delta.trim().length > 0` so whitespace-only strings are consistently treated as no-delta. Resolving.

File: src/lib/StatCard/StatCard.svelte (hunk 3 above)
Comment (murdore) [reply src/lib/StatCard/StatCard.svelte:52]: Acknowledged. StatCard intentionally does not track pressed state — like `Card.svelte`, it is a stateless display primitive. Consumers who need toggle semantics can manage `aria-pressed` externally. Not adding it to avoid over-engineering a component that has no inherent selected/unselected concept. Resolving as WONTFIX.

File: src/wc/components/StatCard.wc.svelte
Comment (murdore) [reply src/wc/components/StatCard.wc.svelte:1]: Fixed in the current HEAD: `deltaPositive` now has `reflect: true` (and `attribute: 'delta-positive'`) in the web component props definition, consistent with other boolean props in the library. Resolving.

File: src/wc/components/StatCard.wc.svelte (hunk 6 above)
Comment (murdore) [reply src/wc/components/StatCard.wc.svelte:11]: Duplicate of the earlier thread — same fix applies. `deltaPositive: { type: 'Boolean', attribute: 'delta-positive', reflect: true }` is in HEAD. Resolving.

File: src/wc/components/StatCard.wc.svelte
Comment (murdore) [reply src/wc/components/StatCard.wc.svelte:1]: Confirmed — snippet forwarding is in place. Resolving this confirmation thread.

File: src/lib/StatCard/StatCard.svelte (hunk 4 above)
Comment (murdore) [reply src/lib/StatCard/StatCard.svelte:26]: Confirmed — delta whitespace handling is correct in HEAD. Resolving.

File: src/lib/StatCard/StatCard.svelte (hunk 5 above)
Comment (murdore) [reply src/lib/StatCard/StatCard.svelte:52]: Confirmed — the `role="button"`, `tabindex`, and `handleKeydown` implementation follows the Card.svelte pattern correctly. Resolving.

File: src/lib/index.ts
Comment (murdore) [reply src/lib/index.ts:1]: Confirmed — StatCard is correctly exported alphabetically and the type export is in place. Resolving.

File: src/lib/StatCard/StatCard.svelte
Comment (murdore) [reply src/lib/StatCard/StatCard.svelte:1]: Acknowledged. The library currently has no per-component test files (Banner, Card, Modal, etc. are all untested at unit level — only `tests/pagination.test.ts` and a placeholder exist). Adding a component test suite is worthwhile as a follow-up initiative for the whole library, but not scoped to this PR. Resolving — the concern stands as a library-wide gap, not a blocker for StatCard specifically.
